File: mycollab-ui/src/main/java/com/esofthead/mycollab/vaadin/mvp/AbstractCssPageView.java
/**
 * This file is part of mycollab-ui.
 *
 * mycollab-ui is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-ui is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-ui.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.vaadin.mvp;

import com.vaadin.ui.VerticalLayout;

import java.io.Serializable;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public abstract class AbstractCssPageView extends VerticalLayout implements PageView, Serializable {
    public AbstractCssPageView() {
        this.setStyleName("page-view");
        this.setSizeFull();
    }

    @SuppressWarnings("rawtypes")
    @Override
    public void addViewListener(ViewListener listener) {
        addListener(ViewEvent.VIEW_IDENTIFIER(), ViewEvent.class, listener, ViewListener.viewInitMethod);
    }
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/mvp/AbstractCssPageView.kt
package com.mycollab.vaadin.mvp

import com.mycollab.vaadin.event.ViewEvent
import com.vaadin.ui.CssLayout

import java.io.Serializable

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
abstract class AbstractCssPageView : CssLayout(), PageView, Serializable {
    init {
        this.setSizeFull()
    }

    override fun <E> addViewListener(listener: PageView.ViewListener<E>) {
        addListener(ViewEvent.VIEW_IDENTIFIER, ViewEvent::class.java, listener, PageView.ViewListener.viewInitMethod)
    }
}
